File: frontend/src/services/careerDevelopmentService.js
import apiClient from './api';

export const getCareerPaths = () => apiClient.get('/suite/career-development/paths');
export const createCareerPath = (dto) => apiClient.post('/suite/career-development/paths', dto);
export const getMentorshipPairs = () => apiClient.get('/suite/career-development/mentorship');
export const createMentorshipPair = (dto) => apiClient.post('/suite/career-development/mentorship', dto);
export const getDevelopmentPlans = () => apiClient.get('/suite/career-development/plans');
export const createDevelopmentPlan = (dto) => apiClient.post('/suite/career-development/plans', dto);

File: backend/src/main/java/com/awais/hr/module/career/service/CareerDevelopmentService.java
package com.awais.hr.module.career.service;

import java.util.List;
import java.util.Map;

public interface CareerDevelopmentService {
    List<Map<String, Object>> getCareerPaths();
    Map<String, Object> createCareerPath(Map<String, Object> body);
    List<Map<String, Object>> getMentorshipPairs();
    Map<String, Object> createMentorshipPair(Map<String, Object> body);
    List<Map<String, Object>> getDevelopmentPlans();
    Map<String, Object> createDevelopmentPlan(Map<String, Object> body);
}
